BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java

```diff
@@ -111,6 +114,120 @@ public boolean onOptionsItemSelected(MenuItem item) {
+      requestStoragePermission();
+    } else {
+      // if the user has already allowed access to device external storage
+      exportBibTex.createBibFile();
+      exportBibTex.writeBibFile(retrieveBibContentShelf());
+
+      Toast.makeText(getContext(),
+          getString(R.string.exported_file_stored_in) + '\n'
+              + File.separator + folderName + File.separator + fileName
+              + fileTypeBibToast, Toast.LENGTH_LONG).show();
+    }
+  }
+
+  private void requestStoragePermission() {
+    if (ActivityCompat.shouldShowRequestPermissionRationale(getActivity(),
+        Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+      showRequestPermissionDialog();
+    } else {
+      requestPermissions(new String[] {Manifest.permission.WRITE_EXTERNAL_STORAGE},
+          STORAGE_PERMISSION_CODE);
+    }
+  }
+
+  private void showRequestPermissionDialog() {
+    AlertDialog.Builder reqAlertDialog = new AlertDialog.Builder(getContext());
+    reqAlertDialog.setTitle(R.string.storage_permission_needed);
+    reqAlertDialog.setMessage(R.string.storage_permission_alert_msg);
+
+    reqAlertDialog.setPositiveButton(R.string.ok,
+        (dialog, which) -> ActivityCompat.requestPermissions(getActivity(),
+            new String[] {Manifest.permission.WRITE_EXTERNAL_STORAGE},
+            STORAGE_PERMISSION_CODE));
+    reqAlertDialog.setNegativeButton(R.string.cancel,
+        (dialog, which) -> dialog.dismiss());
+
+    reqAlertDialog.create().show();
+  }
+
+  /**
+   * Callback method, that checks the result from requesting permissions.
+   *
+   * @param requestCode unique integer value for the requested permission
+   *                    This value is given by the programmer.
+   * @param permissions array of requested name(s)
+   *                    of the permission(s)
+   * @param grantResults grant results for the corresponding permissions
+   *                     which is either PackageManager.PERMISSION_GRANTED
+   *                     or PackageManager.PERMISSION_DENIED.
+   */
+  @Override
+  public void onRequestPermissionsResult(int requestCode,
+                                         String[] permissions, int[] grantResults) {
+
+    switch (requestCode) {
+
+      case STORAGE_PERMISSION_CODE:
+        if (grantResults.length > 0
+            && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+          exportBibTex.createBibFile();
+          exportBibTex.writeBibFile(retrieveBibContentShelf());
+
+          Toast.makeText(getContext(),
+              getString(R.string.exported_file_stored_in) + '\n'
+                  + File.separator + folderName + File.separator + fileName
+                  + fileTypeBibToast, Toast.LENGTH_LONG).show();
+
+        } else {
+          Toast.makeText(getContext(), R.string.storage_permission_denied,
+              Toast.LENGTH_SHORT).show();
+        }
+        break;
+
+      default:
```

Comment:
Sarah said in my review that we should add a break; statement after default. So please add it.
```suggestion
      default:
        break;
```